app/exec/extension/install.ts

```diff
@@ -0,0 +1,489 @@
+import { TfCommand } from "../../lib/tfcommand";
+import args = require("../../lib/arguments");
+import colors = require("colors");
+import extBase = require("./default");
+import extInfo = require("./_lib/extensioninfo");
+import Q = require("q");
+import trace = require("../../lib/trace");
+import GalleryInterfaces = require("vso-node-api/interfaces/GalleryInterfaces");
+import gallerym = require("vso-node-api/GalleryApi");
+
+const SPS_INSTANCE_TYPE = "951917AC-A960-4999-8464-E3F0AA25B381";
+
+export function getCommand(args: string[]): TfCommand<extBase.ExtensionArguments, ExtensionInstallResult> {
+    return new ExtensionInstall(args);
+}
+
+interface TargetAccount {
+    accountName: string;
+    accountId: string;
+}
+
+export class AccountInstallReport {
+    constructor(
+        public itemId: string,
+        public accountName: string,
+        public accountId: string,
+        public installed: boolean = false,
+        public reason?: string) {
+    }
+
+    setError(reason: string) {
+        this.installed = false;
+        this.reason = reason;
+    }
+
+    setInstalled(reason?: string) {
+        this.installed = true;
+        this.reason = reason;
+    }
+}
+
+export class ExtensionInstallResult {
+    constructor(
+        public itemId: string,
+        public accountReports: AccountInstallReport[]) {
+    }
+
+    hasErrors(): boolean {
+        return this.accountReports.some(t => !t.installed);
+    }
+
+    toString() {
+        let result: string;
+        const status: string = this.hasErrors() ? "Failed" : "Succeeded";
+
+        result = `=== Installation ${status} === ` +
+            `\n  - ItemId: ${this.itemId}` +
+            "\n  - Accounts:";
+
+        this.accountReports.forEach(t => {
+            result += `\n    - ${t.accountName}: [${t.installed ? "OK" : `Failed: ${t.reason}`}]`;
+        });
+
+        return result;
+    }
+}
+
+export class ExtensionInstall extends extBase.ExtensionBase<ExtensionInstallResult> {
+    protected description = "Install a Visual Studio Services Extension in a list of VSTS Accounts.";
+
+    constructor(passedArgs: string[]) {
+        super(passedArgs);
+        this.registerCommandArgument("accounts", "Installation target accounts", "List of accounts where to install the extension.", args.ArrayArgument);
```

Comment:
Override setCommandArgs(), call super.setCommandArgs(), then register this new one.